File: java/단계별/약수,_배수와_소수/5086(배수와_약수).java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        // while문 안에 1을 넣으면 안된다.
        while (true) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");

            int A = Integer.parseInt(st.nextToken());
            int B = Integer.parseInt(st.nextToken());

            // !A !B가 작동하지 않는다.
            if (A == 0 && B == 0) {
                break;
            }

            if (B > A && B % A == 0) {
                bw.write("factor\n");
            }
            else if (A > B && A % B == 0) {
                bw.write("multiple\n");
            }
            else {
                bw.write("neither\n");
            }
        }

        br.close();
        bw.flush();
        bw.close();
    }
}

File: 단계별/정수론_및_조합론/5086(배수와_약수).cpp
#include <iostream>

using namespace std;

int main()
{
	while (1) {
		int num1, num2;

		cin >> num1 >> num2;

		if (num1 == 0 && num2 == 0)
			break;

		if (num2 % num1 == 0) {
			cout << "factor\n";
			continue;
		}

		if (num1 % num2 == 0) {
			cout << "multiple\n";
			continue;
		}

		cout << "neither\n";
	}
}